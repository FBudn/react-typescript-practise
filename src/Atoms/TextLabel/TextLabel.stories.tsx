import React from "react";
import { ComponentMeta, ComponentStory } from "@storybook/react";
import TextLabel from "./TextLabel";

export default {
  title: "Atoms/TextLabel",
  component: TextLabel,
} as ComponentMeta<typeof TextLabel>;

const Template: ComponentStory<typeof TextLabel> = (args) => (
  <TextLabel {...args} />
);

export const BasicRememberMe = Template.bind({});
BasicRememberMe.args = {
  children: "Remember me?",
  color: "black",
  align: "center",
  justify: "center",
  fontWeight: "bold",
  fontSize: "1rem",
  width: "100%",
  margin: "2px",
  cursor: "pointer",
};

export const BasicForgotPassword = Template.bind({});
BasicForgotPassword.args = {
  children: "Forgot Password?",
  color: "black",
  align: "center",
  justify: "center",
  fontWeight: "bold",
  fontSize: "1rem",
  width: "100%",
  margin: "2px",
  cursor: "pointer",
};

export const BasicAlreadyUser = Template.bind({});
BasicAlreadyUser.args = {
  children: "Already a user?",
  color: "black",
  align: "center",
  justify: "center",
  fontWeight: "bold",
  fontSize: "1rem",
  width: "100%",
  margin: "2px",
  cursor: "pointer",
};

export const BasicNeedAccount = Template.bind({});
BasicNeedAccount.args = {
  children: "Need an account?",
  color: "black",
  align: "center",
  justify: "center",
  fontWeight: "bold",
  fontSize: "1rem",
  width: "100%",
  margin: "2px",
  cursor: "pointer",
};
