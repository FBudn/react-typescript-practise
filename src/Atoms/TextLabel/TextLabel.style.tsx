import styled from "styled-components";

export const TextLabelContainer = styled.div.attrs(
  (props: {
    color: string;
    align: string;
    justify: string;
    fontWeight: string;
    fontSize: string;
    width: string;
    margin: string;
    cursor: string;
  }) => props,
)`
  display: flex;
  border: none;
  width: ${(props) => props.width};
  align-items: ${(props) => props.align};
  color: ${(props) => props.color};
  justify-content: ${(props) => props.justify};
  font-weight: ${(props) => props.fontWeight};
  font-size: ${(props) => props.fontSize};
  margin: ${(props) => props.margin};
  cursor: ${(props) => props.cursor};
`;
