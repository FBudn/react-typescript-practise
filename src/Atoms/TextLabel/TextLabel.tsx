import React from "react";
import { LabelContainer } from "./TextLabelStyle";

export interface TextLabelProps {
  children: React.ReactNode;
  color: string;
  align: string;
  justify: string;
  fontWeight: string;
  fontSize: string;
  width?: string;
  margin?: string;
  cursor?: string;
}

const TextLabel: React.FC<TextLabelProps> = ({
  children,
  color,
  align,
  justify,
  fontWeight,
  fontSize,
  width,
  margin,
  cursor,
}) => {
  return (
    <LabelContainer
      data-testid="TextLabel-1"
      color={color}
      align={align}
      justify={justify}
      fontWeight={fontWeight}
      fontSize={fontSize}
      width={width}
      margin={margin}
      cursor={cursor}
    >
      {children}
    </LabelContainer>
  );
};

export default TextLabel;
