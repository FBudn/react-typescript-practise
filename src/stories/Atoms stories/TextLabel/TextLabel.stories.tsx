import React from "react";
import { ComponentMeta, ComponentStory } from "@storybook/react";
import TextLabel from "../../../Atoms/TextLabel/TextLabel";

export default {
  title: "Atoms/TextLabel",
  component: TextLabel,
} as ComponentMeta<typeof TextLabel>;

const Template: ComponentStory<typeof TextLabel> = (args) => (
  <TextLabel {...args} />
);

export const BasicRememberMe = Template.bind({});
BasicRememberMe.args = {
  children: "Remember me?",
  color: "black",
  align: "center",
  justify: "center",
  fontWeight: "bold",
  fontSize: "1rem",
  width: "100%",
  margin: "2px",
  cursor: "pointer",
};

export const BasicForgotPassword = Template.bind({});
BasicForgotPassword.args = {
  children: "Forgot Password?",
  color: "black",
  align: "center",
  justify: "center",
  fontWeight: "bold",
  fontSize: "1rem",
  width: "100%",
  margin: "2px",
  cursor: "pointer",
};

export const BasicAlreadyUser = Template.bind({});
BasicAlreadyUser.args = {
  children: "Already a user",
  color: "black",
  align: "center",
  justify: "center",
  fontWeight: "bold",
  fontSize: "1rem",
  width: "100%",
  margin: "2px",
  cursor: "pointer",
};
/* 
import React from "react";
import { Meta, Story } from "@storybook/react";
import Input, { InputProps } from "../../../Atoms/Input/Input";
import TextLabel, { TextLabelProps } from "../../../Atoms/TextLabel/TextLabel";

export default {
  title: "Molecules/InputAndLabel",
  component: Input,
  type: "Text",
} as Meta;

const Template: Story<InputProps, TextLabelProps> = (args) => (
  <>
    <TextLabel
      color="black"
      align="center"
      justify="center"
      fontWeight="bold"
      fontSize="2xl"
    >
      {args.children}
    </TextLabel>
    <Input type={args.type} />
  </>
);

export const Basic = Template.bind({});
Basic.args = {
  children: "Email",
};

*/
