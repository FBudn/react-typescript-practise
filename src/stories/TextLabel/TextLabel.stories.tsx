import React from "react";
import { Meta, Story } from "@storybook/react";
import TextLabel, { TextLabelProps } from "../../Atoms/TextLabel/TextLabel";

export default {
  title: "Atoms/TextLabel",
  component: TextLabel,
  argTypes: {
    onClickLink: { action: "Im switching page" },
  },
} as Meta;

const Template: Story<TextLabelProps> = (args) => (
  <TextLabel {...args}>{args.children}</TextLabel>
);

export const Basic = Template.bind({});
Basic.args = {
  children: "Text label",
  color: "black",
  align: "center",
  justify: "center",
  fontWeight: "bold",
  fontSize: "1rem",
  width: "100%",
  margin: "2px",
  cursor: "pointer",
};
