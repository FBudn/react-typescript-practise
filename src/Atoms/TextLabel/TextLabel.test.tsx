/* eslint-disable react/react-in-jsx-scope */
import { render, screen } from "@testing-library/react";
import TestRenderer from "react-test-renderer";
import TextLabel from "./TextLabel";
import "jest-styled-components";

test(`Should render TextLabel component`, () => {
  render(
    <TextLabel
      color="black"
      align="center"
      justify="center"
      fontSize="1rem"
      fontWeight="500"
    >
      TextLabel Test
    </TextLabel>,
  );
  const TextLabelElement = screen.getByTestId(`TextLabel-1`);
  expect(TextLabelElement).toBeInTheDocument();
});

test(`Should match snapshot and havestyles:`, () => {
  const tree = TestRenderer.create(
    <TextLabel
      color="black"
      align="center"
      justify="center"
      fontSize="1rem"
      fontWeight="500"
    >
      TextLabel Test
    </TextLabel>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule("display", "flex");
  expect(tree).toHaveStyleRule("border", "none");
});
