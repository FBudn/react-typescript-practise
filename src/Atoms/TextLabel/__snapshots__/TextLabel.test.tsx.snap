// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should match snapshot and havestyles: 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  border: none;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: black;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-weight: 500;
  font-size: 1rem;
}

<div
  className="c0"
  color="black"
  data-testid="TextLabel-1"
  fontSize="1rem"
  fontWeight="500"
>
  TextLabel Test
</div>
`;
